### PR TITLE
feat(release): per-releasable changelog co-located with each standing-PR row

### PR DESCRIPTION
**Pull request by goosewobbler**
## What

Closes #488. The standing PR previously appended one combined `<details>` changelog (every package concatenated). Reviewers asking "what changed in THIS unit?" had to expand and scan the whole blob.

Now:
- **Per-row changelog.** Each checkbox row gets its own collapsed `<details>` showing only its changes. A streamlined `primaryPackages` unit aggregates primary + coupled members + changed prerequisites; every other row shows its single package. A prerequisite shared by two units appears under each (self-contained, by design).
- **Hold-back sync.** Holding a package back greys its row's changelog (`<s>…</s> — held back`) and removes it from the footer.
- **Combined footer (flat + deduped).** A single default-collapsed footer lists every change once, grouped by change type (Added/Fixed/Changed), de-duplicated by the underlying commit so a change touching N packages appears once, with an optional inline `(scope)` tag. Config-gated via `ci.standingPr.combinedChangelogFooter` (default on).

## Body-cap safety

Per-row changelogs duplicate prerequisites, so `renderPrBody` degrades in stages under the 64 KB cap: full → drop footer → bare rows + a pointer to `CHANGELOG.md` → bare rows. Rows are never dropped; sync mode truncates the footer at a line boundary.

## For #487

Per-row attachment is a reusable `RowChangelogRenderer` injected into `renderSelectionRegion`. #487 changes *where* rows are placed, not *how* changelogs attach.

## Tests

`changelog-region.spec.ts` (per-unit aggregation, shared-prerequisite duplication, synthetic filtering, hold-back greying, footer dedup) and updated `standing-pr.spec.ts` integration coverage. Full gate green; `pnpm schema:check` clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<details open><summary><h3>Greptile Summary</h3></summary>

This PR replaces the single combined `<details>` changelog blob in the standing PR with per-row changelogs co-located with each releasable checkbox row, plus a new flat, de-duplicated combined footer. The feature adds a new `changelog-region.ts` module that handles all changelog formatting, introduces a `combinedChangelogFooter` config toggle, and graduates the PR body truncation logic to a 4-level degradation strategy.

- **`changelog-region.ts`** is a new module providing `makeRowChangelogRenderer` (factory for per-row `<details>` blocks that aggregate primary + coupled members) and `renderCombinedFooter` (flat, deduped, type-grouped footer driven by the write output so held-back packages are excluded). Both the previously flagged `dedupeKey` scope-collision bug and the `sharedEntries`-invisible-when-footer-disabled bug have been corrected in this version.
- **`standing-pr.ts`** removes the old `renderChangelogSection` / `renderChangelogEntries` helpers and rewires `renderPrBody` to accept a lazily-evaluated `renderSelectionBlock(withChangelogs: boolean)` so the 4-level cap degradation (full → drop footer → bare rows + notice → bare rows) can incrementally strip changelog volume without ever losing the interactive checkbox rows.
</details>

<h3>Confidence Score: 5/5</h3>

Safe to merge — both previously flagged correctness issues (scope in dedupeKey, sharedEntries invisible when footer disabled) are resolved in this version, and the 4-level body-cap degradation preserves the interactive rows at all stages.

The refactoring is well-isolated in the new changelog-region.ts module, the combinedChangelogFooter gate correctly falls back to sharedOnly rendering rather than silently dropping entries, the sync-mode override is explicit and tested, and all edge cases have dedicated unit and integration coverage.

No files require special attention.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/release/src/standing-pr/changelog-region.ts | New module providing per-row and combined-footer changelog rendering; deduplication key correctly includes scope; sharedOnly mode correctly surfaces project-wide entries when full footer is disabled. |
| packages/release/src/standing-pr/standing-pr.ts | Removes old changelog rendering helpers, replaces fixed selectionBlock with re-invokable renderSelectionBlock factory for 4-level body-cap degradation; sync-mode footer bypass logic is correct. |
| packages/release/src/standing-pr/selection-region.ts | Adds optional RowChangelogRenderer parameter; attachChangelog correctly threads per-row changelogs into flat, hierarchical-streamlined, and hierarchical-granular rendering paths. |
| packages/release/test/unit/changelog-region.spec.ts | New spec covering per-unit aggregation, synthetic-entry filtering, hold-back greying, footer dedup, sharedOnly mode, and scope-distinct lines; good coverage of the addressed edge cases. |
| packages/release/test/unit/standing-pr.spec.ts | Integration tests updated with new per-row / combined-footer assertions; adds sync-mode footer override, sharedEntries-with-disabled-footer, and held-back greying scenarios. |
| packages/config/src/schema.ts | Adds combinedChangelogFooter boolean field with default true; schema and description are consistent with behaviour in standing-pr.ts. |
| docs/configuration.md | Documents combinedChangelogFooter; description correctly notes sync-mode override and sharedEntries survival when the footer is disabled. |
| releasekit.schema.json | Adds combinedChangelogFooter to the JSON schema; type, default, and description match schema.ts. |

</details>

<details><summary><h3>Flowchart</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[runStandingPRUpdate] --> B{strategy == sync?}
    B -- No --> C[makeRowChangelogRenderer from dry changelogs]
    C --> D[renderSelectionBlock factory withChangelogs: bool]
    B -- Yes --> E[renderSelectionBlock = undefined]
    D --> F{footerEnabled OR sync?}
    E --> F
    F -- Yes --> G[renderCombinedFooter full]
    F -- No --> H[renderCombinedFooter sharedOnly=true]
    G --> I[renderPrBody]
    H --> I
    I --> J{body <= 64KB?}
    J -- Yes --> K[Return full body Level 0]
    J -- No non-sync --> L{footer non-empty?}
    L -- Yes --> M{drop footer fits?}
    M -- Yes --> N[Return rows + per-row CLs Level 1]
    M -- No --> O[bareSelection]
    L -- No --> O
    O --> P{bare + notice fits?}
    P -- Yes --> Q[Return bare rows + notice Level 2]
    P -- No --> R[Return bare rows only Level 3]
    J -- No sync --> S[truncateAtLineBoundary + TRUNCATION_NOTICE]
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A[runStandingPRUpdate] --> B{strategy == sync?}
    B -- No --> C[makeRowChangelogRenderer from dry changelogs]
    C --> D[renderSelectionBlock factory withChangelogs: bool]
    B -- Yes --> E[renderSelectionBlock = undefined]
    D --> F{footerEnabled OR sync?}
    E --> F
    F -- Yes --> G[renderCombinedFooter full]
    F -- No --> H[renderCombinedFooter sharedOnly=true]
    G --> I[renderPrBody]
    H --> I
    I --> J{body <= 64KB?}
    J -- Yes --> K[Return full body Level 0]
    J -- No non-sync --> L{footer non-empty?}
    L -- Yes --> M{drop footer fits?}
    M -- Yes --> N[Return rows + per-row CLs Level 1]
    M -- No --> O[bareSelection]
    L -- No --> O
    O --> P{bare + notice fits?}
    P -- Yes --> Q[Return bare rows + notice Level 2]
    P -- No --> R[Return bare rows only Level 3]
    J -- No sync --> S[truncateAtLineBoundary + TRUNCATION_NOTICE]
```

</a>
</details>

<sub>Reviews (3): Last reviewed commit: ["fix(release): keep changelog content whe..."](https://github.com/goosewobbler/releasekit/commit/f0f00a5c95d1b942bc536e8160bd7019d0c102fb) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=40657017)</sub>

### CODE REVIEW

File: packages/release/src/standing-pr/changelog-region.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+function entryLine(d: DedupedEntry, attribution: boolean): string {
+  const { entry, pkgs } = d;
+  const scope = entry.scope ? ` (\`${entry.scope}\`)` : '';
+  const issues = entry.issueIds?.length ? ` ${entry.issueIds.join(', ')}` : '';
+  const attr = attribution && pkgs.size > 0 ? ` _(${[...pkgs].map(shortName).sort().join(', ')})_` : '';
+  return `- ${entry.description}${scope}${issues}${attr}`;
+}
+
```
[the remark this answers is not in the document]
Comment (goosewobbler) [reply packages/release/src/standing-pr/changelog-region.ts:108]: Good catch — fixed in 11cd28d9. `dedupeKey` now includes `e.scope ?? null`, so same-description/different-scope commits stay distinct while same-commit dedup across packages is preserved (same commit → same scope). Added a regression test (`should keep same-description changes with different scopes as distinct lines`).

File: packages/release/src/standing-pr/changelog-region.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      for (const entry of cl.entries) attributed.push({ entry, pkg: name });
+    }
+    const deduped = dedupe(attributed);
+    if (deduped.length === 0) return '';
+    const summary = heldBack
+      ? `<s>Changelog (${pluralEntries(deduped.length)})</s> — held back, won’t publish`
+      : `Changelog (${pluralEntries(deduped.length)})`;
+    return wrapDetails(summary, renderGrouped(deduped), indent);
+  };
+}
```
[the remark this answers is not in the document]
Comment (goosewobbler) [reply packages/release/src/standing-pr/changelog-region.ts:182]: Good catch — real gap, fixed in f0f00a5c. The gate was dropping changelog data, not just suppressing the redundant per-package summary. Now: **sync mode always renders the full footer** (it's the sole changelog surface there — no per-row changelogs), and **non-sync with the footer off renders project-wide `sharedEntries`** via a new `sharedOnly` mode on `renderCombinedFooter` ('Show project-wide changes (N)'), so the otherwise-homeless shared entries survive while the per-package dedup (covered per-row) stays suppressed. Corrected the config `.describe()` (regenerated schema + docs) and added tests for sync+off, non-sync+off+shared, and non-sync+off+no-shared.
